torchtext/datasets/amazonreviewpolarity.py

```diff
@@ -40,11 +40,25 @@ def AmazonReviewPolarity(root: str, split: Union[Tuple[str], str]):
         raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
 
     url_dp = IterableWrapper([URL])
-    cache_dp = url_dp.on_disk_cache(
+    cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
-    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
-    cache_dp = FileOpener(cache_dp, mode="b")
-    extracted_files = cache_dp.read_from_tar()
-    filter_extracted_files = extracted_files.filter(lambda x: _EXTRACTED_FILES[split] in x[0])
-    return filter_extracted_files.parse_csv().map(fn=lambda t: (int(t[0]), ' '.join(t[1:])))
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    def extracted_filepath_fn(x):
+        file_path = os.path.join(root, _EXTRACTED_FILES[split])
+        dir_path = os.path.dirname(file_path)
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        return file_path
+
+    cache_decompressed_dp = cache_compressed_dp.on_disk_cache(
+        filepath_fn=extracted_filepath_fn)
+    cache_decompressed_dp = FileOpener(cache_decompressed_dp, mode="b").\
+        read_from_tar().\
+        filter(lambda x: _EXTRACTED_FILES[split] in x[0]).\
+        map(lambda x: (x[0].replace('_PATH' + '/', ''), x[1]))
```

Comment:
Should this be `f'{_PATH}'` instead of just `'_PATH'`

Reply:
Ahh, that's a good catch, and I just realized it's not necessary to do this since I used `same_filepath_fn=True` in end_caching. So I guess we can remove this mapping of path altogether.